sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/socket.scala

```diff
@@ -136,7 +149,8 @@ class TextSocketSourceProvider extends StreamSourceProvider with DataSourceRegis
       parameters: Map[String, String]): Source = {
     val host = parameters("host")
     val port = parameters("port").toInt
-    new TextSocketSource(host, port, sqlContext)
+    new TextSocketSource(host, port,
+      parameters.getOrElse("includeTimestamp", "false").toBoolean, sqlContext)
```

Comment:
will throw error if includeTimestamp is not parseable as boolean. 
also why is this condition check in two places? there should be one place checking the option and handling parsing errors and all.